docs/developer/architecture/alerting.rst

```diff
@@ -43,3 +43,295 @@ maintenance operation.
 
 .. _here: https://prometheus.io/docs/alerting/configuration/
+
+Design Choices
+--------------
+
+To be able to edit existing rules, add new ones, etc., and in ordrer to keep
+these changes across restorations, upgrades and downgrades, we need to put in
+place some mechanisms to configure Prometheus and Alertmanager and persist
+these configurations.
+
+For the persistence part, we will rely on what has been done for
+:doc:`CSC<configurations>` (Cluster and Services Configurations), and use
+the already defined resources for :term:`Alertmanager` and :term:`Prometheus`.
+
+Extra Prometheus Rules
+~~~~~~~~~~~~~~~~~~~~~~
+
+We will use the already existing ``metalk8s-prometheus-config``
+:term:`ConfigMap` to store the :term:`Prometheus` configuration
+customizations.
+
+Adding extra alert and record rules will be done editing this :term:`ConfigMap`
+under the ``spec.extraRules`` key in ``config.yaml`` as follows::
+
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: metalk8s-prometheus-config
+      namespace: metalk8s-monitoring
+    data:
+      config.yaml: |-
+        apiVersion: addons.metalk8s.scality.com
+        kind: PrometheusConfig
+        spec:
+          deployment:
+            replicas: 1
+          extraRules:
+            groups:
+            - name: <rulesGroupName>
+              rules:
+              - alert: <AlertName>
+                annotations:
+                  description: description of what this alert is
+                expr: vector(1)
+                for: 10m
+                labels:
+                  severity: critical
+              - alert: <AnotherAlertName>
+                [...]
+              - record: <recordName>
+                [...]
+            - name: <anotherRulesGroupName>
+              [...]
+
+PromQL_ is to be used to define ``expr`` field.
+
+This ``spec.extraRules`` entry will be used to generate through Salt a
+``PrometheusRule`` object named ``metalk8s-prometheus-extra-rules`` in the
+``metalk8s-monitoring`` namespace, which will be automatically consumed by the
+:term:`Prometheus` Operator to generate the new rules.
+
+A CLI and UI tooling will be provided to show and edit this configuration.
+
+.. _PromQL: https://prometheus.io/docs/prometheus/latest/querying/basics/
+
+Edit Existing Prometheus Alert Rules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To edit existing :term:`Prometheus` rules, we can't only define new
+``PrometheusRules`` resources since :term:`Prometheus` Operator will not
+overwrite those already existing, but will rather append them to the list of
+rules, ending up with 2 rules with the same name but different parameters.
+
+We also can't edit the ``PrometheusRules`` deployed by MetalK8s, otherwise we
+would lose these changes in case of cluster restoration, upgrade or downgrade.
+
+So, in order to allow the user to customize the alert rules, we will pick up
+some of them (the most relevant ones) and expose only few parts of their
```

Comment:
@alexandre-allard-scality So you expect me and TS to define this specific list of rules one Platform Administrator should be able to customize?

Reply:
Yes, I was planning to do a meeting with TS and you (not sure when, but this week would be nice).
I already extracted the list of available rules, so maybe I just need to put it in an excel sheet, then everybody involved can start thinking about it.